src/components/graph/Graph.jsx

```diff
@@ -235,14 +235,27 @@ export default class Graph extends React.Component {
       draggedNode.oldX = draggedNode.x;
       draggedNode.oldY = draggedNode.y;
 
-      draggedNode.x += d3Event.dx;
-      draggedNode.y += d3Event.dy;
-
-      // set nodes fixing coords fx and fy
-      draggedNode["fx"] = draggedNode.x;
-      draggedNode["fy"] = draggedNode.y;
-
-      this._tick({ draggedNode });
+      const { transform, config } = this.state;
+      const newX = draggedNode.x + d3Event.dx;
+      const newY = draggedNode.y + d3Event.dy;
+      const invertTransformZoom = 1 / transform.k;
+      const shouldUpdateNode =
+        !config.bounded ||
+        (newX > -transform.x * invertTransformZoom &&
+          newX < (config.width - transform.x) * invertTransformZoom &&
+          newY > -transform.y * invertTransformZoom &&
+          newY < (config.height - transform.y) * invertTransformZoom);
```

Comment:
I think this small pice deserves a function on its own. What do you think?

```suggestion
// function should me moved into graph.helpers.js
const isNodeOutOfBounds = (newX, newY, transform, config) => newX < -transform.x * invertTransformZoom &&
          newX > (config.width - transform.x) * invertTransformZoom &&
          newY < -transform.y * invertTransformZoom &&
          newY > (config.height - transform.y) * invertTransformZoom);

      const shouldUpdateNode = !config.bounded || !isNodeOutOfBounds(newX, newY, transform, config);
```

Reply:
I slightly changed the API to only pass position and state, what do you think?